packages/testing/src/fixture.ts

```diff
@@ -126,13 +185,51 @@ export class Fixture {
     }
   }
 
-  async runBinary(cwd?: string) {
+  /** Runs @netlify/build main function programmatic with the provided flags  */
+  async runBuildProgrammatic(): Promise<object> {
+    const { default: build } = await import('@netlify/build')
+    return await build(this.getBuildFlags())
+  }
+
+  async runWithBuild(): Promise<string> {
+    const { default: build } = await import('@netlify/build')
+    const { logs } = await build(this.getBuildFlags())
+    return [logs.stdout.join('\n'), logs.stderr.join('\n')].filter(Boolean).join('\n\n')
+  }
+
+  // TODO: provide better typing if we know what's possible
+  async runDev(devCommand: unknown): Promise<string> {
+    const { startDev } = await import('@netlify/build')
+    const entryPoint = startDev.bind(null, devCommand)
+    const { logs } = await entryPoint(this.getBuildFlags())
+    return [logs.stdout.join('\n'), logs.stderr.join('\n')].filter(Boolean).join('\n\n')
+  }
+
+  /** use the CLI entry point instead of the Node.js main function */
+  runConfigBinary(cwd?: string) {
+    return this.runBinary(this.configBinaryPath, cwd, this.getConfigFlags())
+  }
+
+  /** use the CLI entry point instead of the Node.js main function */
+  runBuildBinary(cwd?: string) {
+    return this.runBinary(this.buildBinaryPath, cwd, this.getBuildFlags())
+  }
+
+  private async runBinary(
+    binary: string,
+    cwd?: string,
+    flags: Record<string, unknown> = {},
```

Comment:
We could be more specific with the flags if we know the types of some of the flags (like env) for example:

```ts
type Flags = {
   [key: string]: unknown,
  env: Record<string, string>,
}
```

Reply:
but it is still in contained in the flags

Reply:
And we could get rid of the cast on line 232

Reply:
updated the PR with it